src/api/users.js

```diff
@@ -269,9 +270,17 @@ export async function updateUserByToken(ctx, token) {
   }
 
   try {
-    await UserModelAPI.findOneAndUpdate({token}, userUpdateObj)
-    ctx.body = 'Successfully set new user password.'
-    return logger.info(`User updated by token ${token}`)
+    const userToBeUpdated = await UserModelAPI.findOne({token})
```

Comment:
If this is for password reset, then we should add a restriction for the users that have "local" provider, right? Same goes for the profile update.

Reply:
I can see that a user created using the openid can have a password also, why not ?